src/Providers/Ollama/Ollama.php

```diff
@@ -94,7 +94,7 @@ public function generateToolsPayload(): array
     protected function createToolCallMessage(array $message): Message
     {
         $tools = \array_map(fn (array $item) => $this->findTool($item['function']['name'])
-            ->setInputs($item['function']['arguments']), $message['tool_calls']);
+            ?->setInputs($item['function']['arguments']), $message['tool_calls']);
```

Comment:
I had to fix this small but unrelated with the PR because it was breaking on my machine.

Reply:
What's the code of the test? Because it's correct that it fails.

Reply:
The tests of this PR.

If this is intended to fail, then ideally we should catch the exception and raise a proper `ProviderException` with a meaningful message.

Reply:
Ok, I chained the AgentException at the Agent level so it should be catch all other underlying exceptions from providers and other components.
 
https://github.com/inspector-apm/neuron-ai/blob/1.10.7/src/HandleChat.php#L64

Reply:
Good, I removed my change.